### PR TITLE
scripts/update-image: fix lint errors

### PR DESCRIPTION
**Pull request by 64**
Fix lint errors according to `pyright`.

Possibly related to managarm/bootstrap-managarm#103.

### CODE REVIEW

File: scripts/update-image.py [hunk 1]
```diff
@@ -9,21 +9,21 @@
...
-@attr.s
+	def __init__(self, blockdev, mount_using, image, mountpoint, root_idx, efi_idx, root_uuid):
+		self.blockdev = blockdev
+		self.mount_using = mount_using
+		self.image = image
+		self.mountpoint = mountpoint
+		self.root_idx = root_idx
+		self.efi_idx = efi_idx
+		self.root_uuid = root_uuid
+
```
Comment (ArsenArsen) [inline scripts/update-image.py:21]: No, attr.ib is used for a reason.

File: scripts/update-image.py [hunk 2]
```diff
@@ -9,21 +9,21 @@
...
 class Partition:
-	idx = attr.ib()
-	type = attr.ib()
-	uuid = attr.ib()
+	def __init__(self, idx, type, uuid):
+		self.idx = idx
+		self.type = type
+		self.uuid = uuid
```
Comment (ArsenArsen) [inline scripts/update-image.py:26]: Likewise (is this missing an `@attr.s`?)

File: scripts/update-image.py [hunk 3]
```diff
@@ -299,9 +300,9 @@ def run(self):
 			# Determine root uuid manually for mountpoint
 			diskdev, partdev = dev_for_mountpoint(self.mountpoint)
 
-			if os.isdir('/dev/disk/by-uuid'):
+			if os.path.isdir('/dev/disk/by-uuid'):
```
Comment (ArsenArsen) [inline scripts/update-image.py:303]: `import os.path as path` is a common convention.

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
Furthermore, rather than these faulty lints we should consider applying `flake8` and some plugins to all our Python code, to enforce PEP8 guidelines and some other QA (import order, complexity requirements).

File: scripts/update-image.py (hunk 1 above)
Comment (64) [reply scripts/update-image.py:21]: I get that. But AFAICT the only feature we use is __init__. So I've just removed attrs entirely from this script.

**Comment by 64** [comment]
I agree, we should really use PEP8.

File: scripts/update-image.py (hunk 3 above)
Comment (64) [reply scripts/update-image.py:303]: I can replace all instances of `os.path` -> `path`, if that's what you're suggesting?

**Comment by avdgrinten** [comment]
I'm fine with keeping `os.path`. I'm indifferent about `attr.s`; if something can be achieved without an external lib, that's probably better anyway. @ArsenArsen what is the real advantage of `attr.s` here, other than not having to write the constructor?

**Comment by ArsenArsen** [comment]
Here? Nothing specifically, this just seemed like a naive lint. If the dep was there already, and it'd appear that it was, it's probably fine to keep it, though.

As for `path`, I'm just mentioning a widespread convention.

**Comment by avdgrinten** [comment]
Why did the linter complain about `attr.s`?

**Comment by ArsenArsen** [comment]
Don't know, I'm just going off the initial comment:

> Fix lint errors according to `pyright`.

**Comment by 64** [comment]
> Why did the linter complain about `attr.s`?

To be clear: it's definitely a false positive lint. The decorator seems to confuse it. There's nothing actually broken about attrs here, unlike the other lint errors.

**Comment by avdgrinten** [comment]
Alright, let's merge this now since there are no hard issues.

**Pull request merged**